### PR TITLE
[MAINTENANCE] Update docs create version script to omit patch

### PR DESCRIPTION
**Pull request by tyler-hoffman**
See title. We only want to cut major.minor versions of docs, so that we don't end up with e.g. 0.18.0, 0.18.1, 0.18.2, etc in versioned_code. We will manage the display of the patch version in the docusaurus config.

This also required updating some tests. I deleted one test, rather than updating it, because the function under test (like many in this file) will be deleted in the very near future.

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: docs/prepare_prior_versions.py [hunk 1]
```diff
@@ -534,20 +534,15 @@ def prepare_prior_version(version: Version) -> None:
     if version >= Version(0, 16):
         prepend_version_info_to_name_for_md_images(version)
 
-    # TODO: delete this after we cut 0.18.9
```
Comment (tyler-hoffman) [inline docs/prepare_prior_versions.py:537]: We've already done this processing, so this is unneeded. (Really most of this file is, but we can address that separately)

File: docs/docs_build.py [hunk 2]
```diff
@@ -86,6 +86,7 @@ def create_version(self, version: Version) -> None:
         os.chdir("..")  # TODO: none of this messing with current directory stuff
         prepend_version_info_to_name_for_snippet_by_name_references(version)
         os.chdir("docusaurus")
+        self._write_release_version(old_version_file)
```
Comment (tyler-hoffman) [inline docs/docs_build.py:83]: Stray fix - sorry for the scope creep. See above for where we store the old version, then temporarily overwrite it.

File: docs/prepare_prior_versions.py [hunk 3]
```diff
@@ -534,20 +534,15 @@ def prepare_prior_version(version: Version) -> None:
...
 @dataclass(frozen=True)
 class Version:
     major: int = 0
     minor: int = 0
-    patch: int = 0
 
-    def as_tuple(self) -> tuple[int, int, int]:
-        return self.major, self.minor, self.patch
+    def as_tuple(self) -> tuple[int, int]:
+        return self.major, self.minor
```
Comment (cdkini) [inline docs/prepare_prior_versions.py:545]: Sorry just seeing this code now - could we instead just leverage `version.Version`? Feels like a place where the stdlib has what we need

**Review by cdkini — APPROVED** [review]
